src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -422,14 +476,214 @@ export class DeployCommand implements Command {
+        });
+      } else {
+        if (deploymentParameter.isExpression) {
+          paramValue = await this.selectValueForParameterOfTypeExpression(
+            _context,
+            paramName,
+            deploymentParameter.value
+          );
+        } else {
+          const options = {
+            title: `Parameter: ${paramName}`,
+            value: deploymentParameter.value,
+            placeHolder: `Please enter value for parameter "${paramName}"`,
+          };
+          paramValue = await vscode.window.showInputBox(options);
+        }
+      }
+
+      if (paramValue != undefined) {
+        const updatedDeploymentParameter: BicepUpdatedDeploymentParameter = {
+          name: paramName,
+          value: paramValue,
+          isSecure: deploymentParameter.isSecure,
+          parameterType: deploymentParameter.parameterType,
+        };
+        updatedDeploymentParameters.push(updatedDeploymentParameter);
+      }
+    }
+
+    return [
+      bicepDeploymentParametersResponse.parametersFileName,
+      updatedDeploymentParameters,
+    ];
+  }
+
+  private async updateParametersFile(
+    _context: IActionContext,
+    documentPath: string,
+    parametersFileExists: boolean,
+    parametersFileName: string
+  ) {
+    let placeholder: string;
+    let parametersFileUpdateOptionString = "None";
+    let parametersFileUpdateOption: ParametersFileUpdateOption;
+    if (parametersFileExists) {
+      parametersFileUpdateOptionString = "Update";
+      parametersFileUpdateOption = ParametersFileUpdateOption.Update;
+      placeholder = `Update ${parametersFileName} with values used in this deployment?`;
+    } else {
+      const folderContainingSourceFile = path.dirname(documentPath);
+      const parametersFilePath = path.join(
+        folderContainingSourceFile,
+        parametersFileName
+      );
+      if (fse.existsSync(parametersFilePath)) {
+        parametersFileUpdateOptionString = "Overwrite";
+        parametersFileUpdateOption = ParametersFileUpdateOption.Overwrite;
+        placeholder = `File ${parametersFileName} already exists. Do you want to overwrite it?`;
+      } else {
+        parametersFileUpdateOptionString = "Create";
+        parametersFileUpdateOption = ParametersFileUpdateOption.Create;
+        placeholder = `Create parameters file from values used in this deployment?`;
+      }
+    }
+
+    const result: IAzureQuickPickItem = await _context.ui.showQuickPick(
+      this._yesNoQuickPickItems,
+      {
+        canPickMany: false,
+        placeHolder: placeholder,
+        suppressPersistence: true,
+      }
+    );
+
+    _context.telemetry.properties.parametersFileUpdateOption =
+      parametersFileUpdateOptionString;
+    if (result == this._yes) {
+      return parametersFileUpdateOption;
+    } else {
+      return ParametersFileUpdateOption.None;
+    }
+  }
+
+  private async selectValueForParameterOfTypeExpression(
+    _context: IActionContext,
```

Comment:
Why are you handling literal default values differently than expressions?  The prompt is much different than the others and is confusing.  Plus it's unnecessary - let's just show the same experience for defaults that use expressions.

Reply:
May need to run it by Uche.

Reply:
For string expression, we currently show:
![image](https://user-images.githubusercontent.com/30270536/172178582-9cd98dc7-b372-4284-a31b-a80eb9373e54.png)
For integer expression, we show:
![image](https://user-images.githubusercontent.com/30270536/172178765-23b19e58-0759-4087-8e32-42a53b8afabc.png)

@ucheNkadiCode and I have discussed this before. I might have missed mentioning the below to you earlier. The reason we chose this design:
If the user chose option (1), we won't update(or provide an option) parameters files. In case of (2) and (3), we will write to parameters file, if user selected update/create option.

We want this separation to make sure expressions don't get written into parameters file. Now about the leave blank option, user can also select option (2) and press enter to get same results.

Let me know if the wordings need to be updated or if there's a use case where this won't work.

Reply:
Thanks @ucheNkadiCode. Removed the "leave blank" option here:
https://github.com/Azure/bicep/commit/5beb12ea871df5d4579492bc7d6c83043d6258b3

Reply:
@ucheNkadiCode 
> However, we run into problems if we allow the user to enter "resourceGroup().location" into the text input and try to interpret it from there. So I'd say leave this as is but get rid of the leave blank option

That's not the suggestion.  The question is, why don't we allow the user to actually tell the params file to use the default value (by not defining the parameter's value in the param file) for literal defaults as well as expression defaults?  Right now you're forcing *current* default value in the bicep file to always be written into the params file.  Thus if the bicep file changes to a different default value, it won't get picked up.

My preference would be to allow the option of keeping the default value for both expression and literals.  Right now you only allow it for expressions.

This is similar to the discussion about whether to place parameters with default values into the params file with the "generate parameters" command.  I guess we'll get feedback.